pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -421,6 +423,21 @@ func (c *ClusterConfig) AppendAvailabilityZone(newAZ string) {
 	c.AvailabilityZones = append(c.AvailabilityZones, newAZ)
 }
 
+// CleanupSubnets clean up subnet entries having invalid AZ
+func (c *ClusterConfig) CleanupSubnets() {
```

Comment:
Since this is a very specific thing for subnets and it's an implementation detail I think I would move it to the vpc.go file, moving the receiver to an argument.

Reply:
@sayboras I think I prefer the way the tests were before. What I was looking for was something like this:
```
// For corner cases test closer to the function
testCleanup() {
  c = ClusterConfig{ .... }  // test data
  CleanupSubnets(c)
  Expect(c.subnets).To(//not have duplicates)
}

// One regression test for the public API that makes sure we don't have this issue again
testImport() {
// Mock EC2 calls
  p.MockEC2().On("DescribeVpcsInput", MatchedBy(func(input *ec2.DescribeVpcsInput) bool {
      return input != nil
  })).Return(...)

  ImportAllSubnets(p, spec) // Use the mocked provider
  
  Expect(spec).To(// not have duplicates...)
}
```
What do you think? At the top of the tests for "VPC - Cluster Endpoints" you can see an example on how we use the EC2 mocks.

Reply:
It's great idea, I should have thought about it earlier. Maybe I just want to understand and test the whole thing, which makes my mind so narrowed :cry: